### PR TITLE
Refactor - ComputeCluster Bootstraps + ENV vars

### PR DESCRIPTION
**Pull request by RossMurr4y**
## Description

A number of minor changes surrounding the implementation of a computecluster and its processing of secrets, linked component attributes and solution attributes into environment settings:
1. Refactor the computecluster link processing for dbs.
2. New bootstrap for installing docker on debian.
3. Update quotations on `stage` bootstrap.
4. Remove "encode" bootstrap + update init-encoded to just "init"

## Motivation and Context

1. both Attributes and Settings on the linked DB may be something that needs to be passed as an ENV variable to the compute cluster. This needed to be handled better. Additionally any that happen to also be secrets need to have the appropriate reference lookups created.
2. This was missing so dockerfile could not be handled.
3. Missing quotations on the connection string caused the shell to misinterpret arguments.
4. There are two ways to exec code via a VM ScaleSet Extension - a `script` path and an `commandToExecute` argument. Previously the bootstraps were passing the init file to the `script` path, and ARM would only execute it if it was base64 encoded - so we did that first and passed the new script. Because `script` and `commandToExecute` are mutually exclusive and all the other bootstraps concatenate to make one long command, script now gets appended to the end of `commandToExecute` and no longer requires encoding.

## How Has This Been Tested?

Local template generation and deployment.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions

- [x] None

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] None of the above.

### CODE REVIEW

File: azure/components/computecluster/setup.ftl [hunk 1]
```diff
@@ -130,8 +130,34 @@
...
+                        [#local dbSecrets = getSettingSecrets(linkTargetSettings, "ENV") + 
+                            [{"DB_PASSWORD" : linkTargetAttributes["SECRET"]}]]
+
+                        [#-- ENV Attributes --]
+                        [#local dbDetails += 
+                            { 
+                                "DB_NAME" : linkTargetAttributes["DB_NAME"],
+                                "DB_USERNAME" : linkTargetAttributes["USERNAME"]
+                            } ]
+[@debug message="flibs" context=dbDetails enabled=true /]
```
Comment (roleyfoley) [inline azure/components/computecluster/setup.ftl:158]: Deflibulation required

File: azure/components/computecluster/setup.ftl [hunk 2]
```diff
@@ -130,8 +130,34 @@
 
             [#case DB_COMPONENT_TYPE]
 
-                [#local dbSecrets = getSettingSecrets(linkTargetSettings, "ENV") +
-                    [{ "DB_PASSWORD" : linkTargetAttributes["SECRET"] }] ]
+                [#local dbSecrets = []]
+
+                [#switch linkTargetConfiguration.Solution.Engine]
```
Comment (roleyfoley) [inline azure/components/computecluster/setup.ftl:135]: Should this be part of the dB component attributes? Would you expect the same to be applied for functions or other components that use dB connections

File: azure/inputsources/shared/masterdata.ftl [hunk 3]
```diff
@@ -1142,13 +1142,22 @@
             }
           }
         },
+        "docker-debian" : {
+          "Index" : 16,
+          "ProtectedSettings" : {
+            "exec" : {
+              "Key" : "commandToExecute",
+              "Value" : "sudo apt install docker.io -y'"
```
Comment (roleyfoley) [inline azure/inputsources/shared/masterdata.ftl:1150]: Does this need to include apt-get update to refresh the apt dB?

Comment (RossMurr4y) [reply azure/inputsources/shared/masterdata.ftl:1150]: That's performed by the `update-debian` boot strap, so it doesn't have to run a zillion times.

File: azure/components/computecluster/setup.ftl (hunk 1 above)
Comment (RossMurr4y) [reply azure/components/computecluster/setup.ftl]: Deflibbed, as requested.

File: azure/components/computecluster/setup.ftl (hunk 2 above)
Comment (RossMurr4y) [reply azure/components/computecluster/setup.ftl:137]: Yes, good point. I'll implement that on the Azure provider now and push up the changes.

Comment (RossMurr4y) [reply azure/components/computecluster/setup.ftl:135]: @roleyfoley  I have updated the PR with commit 8eefc17 to include secret support on the component. I have written it in such a way that we can add Secret Store support at a later time and migrate it to the shared component with minimal effort.

**Pull request merged**